ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/InvokeButton/InvokeModal.tsx

```diff
@@ -0,0 +1,98 @@
+import { useState } from 'react';
+import { Button } from '@inngest/components/Button';
+import { Modal } from '@inngest/components/Modal';
+
+import { Alert } from '@/components/Alert';
+import CodeEditor from '@/components/Textarea/CodeEditor';
+
+const initialCode = { data: {} };
+
+type Props = {
+  isOpen: boolean;
+  onCancel: () => void;
+  onConfirm: (payload: { data: Record<string, unknown> }) => void;
+};
+
+export function InvokeModal({ isOpen, onCancel, onConfirm }: Props) {
+  const [error, setError] = useState<string>();
+
+  function onSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const formData = new FormData(event.currentTarget);
+
+    try {
+      const payload = parseCode(formData.get('code'));
+      onConfirm(payload);
+      setError(undefined);
+    } catch (error) {
+      if (!(error instanceof Error)) {
+        setError('Unknown error');
+        return;
+      }
+
+      setError(error.message);
+    }
+  }
+
+  return (
+    <Modal
+      className="w-full max-w-3xl"
+      description="Invoke this function, triggering a function run"
+      isOpen={isOpen}
+      onClose={onCancel}
+      title="Invoke Function"
+    >
+      <form onSubmit={onSubmit}>
+        <div className="m-6">
+          <CodeEditor
+            className="rounded-lg bg-slate-900 px-4"
+            initialCode={JSON.stringify(initialCode, null, 2)}
+            language="json"
+            name="code"
+          />
+
+          {error && (
+            <Alert className="mt-6" severity="error">
+              {error}
+            </Alert>
+          )}
+        </div>
+
+        <div className="m-6 border-b border-slate-200"></div>
```

Comment:
This is a very minor thing, but if you pass the buttons to the `footer` prop of the Modal, you already get the border